### PR TITLE
[GH-24843]: Fix customize_your_experience overlay position

### PR DESCRIPTION
**Pull request by umrkhn**
#### Summary
This PR fixes  `overlayPunchOut` position as shown in the screenshots below.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24843
Jira https://mattermost.atlassian.net/browse/MM-54744

#### Screenshots
|  before  |  after  |
|----|----|
before | after

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by umrkhn** [comment]
> @cwarnermm can you please tag reviewers for this PR ? Thanks

Although the issue was fixed but this PR goes further and fixes the overlay position as shown in the screenshots above. let me know if it's required or I should close it.

**Comment by umrkhn** [comment]
Thanks @cwarnermm appreciate the help :+1:

**Review by matthewbirtch — APPROVED** [review]
Nice improvement. Thanks @umrkhn!

File: webapp/channels/src/components/global_header/right_controls/right_controls.tsx [hunk 1]
```diff
@@ -39,7 +39,11 @@ const RightControlsContainer = styled.div`
         margin-left: 8px;
     }
 `;
-
+const CustomizeYourExperienceTourWrapper = styled.div`
+    display: flex;
+    align-items: center;
+    height: 100%
+`;
 export type Props = {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/global_header/right_controls/right_controls.tsx:47]: nit empty line inbetween

File: webapp/channels/src/components/global_header/right_controls/right_controls.tsx [hunk 2]
```diff
@@ -70,7 +72,20 @@ const RightControls = ({productId = null}: Props): JSX.Element => {
...
+            <CustomizeYourExperienceTourWrapper id='CustomizeYourExperienceTour'>
+                {
+                    isChannels(productId) ? (
+                        <>
+                            <SettingsButton/>
+                            {showCustomizeTip && <CustomizeYourExperienceTour/>}
+                        </>
+                    ) : null
+                }
+                <div style={{marginLeft: '8px'}}>
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/global_header/right_controls/right_controls.tsx:84]: can we replace this inline styling

File: webapp/channels/src/components/global_header/right_controls/right_controls.tsx [hunk 3]
```diff
@@ -70,7 +72,20 @@ const RightControls = ({productId = null}: Props): JSX.Element => {
...
+                    <StatusDropdown/>
+                </div>
+            </CustomizeYourExperienceTourWrapper>
+
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/global_header/right_controls/right_controls.tsx:88]: nit empty line removal from here

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
thank you @umrkhn

**Review by yasserfaraazkhan — APPROVED** [review]
Tested this on Mac Chrome, desktop and win desktop.
working as expected.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @umrkhn!

**Comment by devinbinnie** [comment]
@umrkhn Can you fix this lint issue and then we can merge?
```
/home/runner/work/mattermost/mattermost/webapp/channels/src/components/global_header/right_controls/right_controls.tsx
Error:   51:2  error  Missing semicolon  semi
```

**Review by devinbinnie — APPROVED** [review]

**Comment by hanzei** [comment]
@devinbinnie @M-ZubairAhmed Can we merge this PR?

**Pull request merged**